### PR TITLE
Search Form UI

### PR DESCRIPTION
**Pull request by noranda**
This PR:
- Adds the search form UI, including all Material UI form fields.
- Pulls in the patient data to autofill `zipcode` and `age` as well as a default travel distance of 100 miles.
- Sets up FHIR requests for user, patient, and patient cancer conditions. 
- Styles components using Material UI properties.
- Adds tests.

TODO:
- Create a FHIR patient to test app with (none of the test patients have cancer that I can tell).
- Hook up remaining fields to appropriate data from the patient record.
- Hook up matching service (currently using mock results data).
- Build out results UI.

TESTING:
- Test all form fields including clearing values.
- Test responsiveness of design (large to mobile size).
- Test multiple patients and users to make sure data is populated correctly.

### CODE REVIEW

File: src/components/SearchForm/SearchForm.tsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+              </Box>
+
+              <Box color="grey.600" fontSize={{ xs: 20, lg: 25, xl: 28 }} fontWeight={300}>
+                Search with data populated from your record, or change to find matching trials
+              </Box>
+            </Box>
+          </Stack>
+        </Box>
+
+        <Grid columns={8} container spacing={2} px={2} py={{ xs: 0, md: 2 }} mt={{ xs: 0, md: -2 }}>
```
Comment (jtquach1) [inline src/components/SearchForm/SearchForm.tsx:73]:
```suggestion
        <Grid columns={8} container spacing={2} px={2} py={{ md: 2 }} pb={{ xs: 2 }} mt={{ xs: 0, md: -2 }}>
```
Maybe, just so the search button has some space before the end of the page? On medium/large screens it appears that there's a bottom padding of 2.

Current: 
image

With suggestion:
image

**Review by jtquach1 — CHANGES_REQUESTED** [review]
Besides that, everything else looks good to me!

File: src/components/SearchForm/FormFields.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  field: ControllerRenderProps<SearchFormValuesType, 'cancerSubtype'>;
+}): ReactElement => (
+  <TextField data-testid="cancerSubtype" fullWidth label="Cancer Subtype" variant="filled" {...field} />
+);
+
+export const MetastatsisTextField = ({
+  field,
+}: {
+  field: ControllerRenderProps<SearchFormValuesType, 'metastasis'>;
+}): ReactElement => <TextField data-testid="metastasis" fullWidth label="Metastatsis" variant="filled" {...field} />;
```
[the remark this answers is not in the document]
Comment (noranda) [reply src/components/SearchForm/FormFields.tsx:56]: Yep! Good catch!

File: src/components/SearchForm/SearchForm.tsx (hunk 1 above)
Comment (noranda) [reply src/components/SearchForm/SearchForm.tsx:73]: I like it!

**Comment by noranda** [comment]
@jtquach1 All issues resolved!

**Review by dmpotter44 — APPROVED** [review]
I haven't really gone through these too closely beyond making sure I can make it run locally (which ran into some VPN issues initially).

**Review by jtquach1 — APPROVED** [review]

**Pull request merged**